### PR TITLE
test(negative): implement node network disconnection

### PR DESCRIPTION
**Pull request by yangchiu**
test: implement node network disconnection

For https://github.com/longhorn/longhorn/issues/6704

Signed-off-by: Yang Chiu <yang.chiu@suse.com>

### CODE REVIEW

**Comment by yangchiu** [comment]
blocked by https://github.com/longhorn/longhorn-tests/pull/1542

**Review by c3y1huang — CHANGES_REQUESTED** [review]
There are overlappings with https://github.com/longhorn/longhorn-tests/pull/1542. I will review this again after that one is merged.

**Comment by yangchiu** [comment]
@longhorn/qa Please help review this. Many thanks.

File: e2e/libs/volume/volume.py
Comment (khushboo-rancher) [inline e2e/libs/volume/volume.py:53]: I assume, making sequence change here makes more sense :D

File: e2e/tests/network_disconnect.robot
Comment (khushboo-rancher) [inline e2e/tests/network_disconnect.robot:29]: Just curious, can we rename test name little bit shorter like `Disconnect Volume Node Network For Long Time While Workload Heavy Writing` and we can add a description to test to explain the pod eviction timeout

**Review by khushboo-rancher — COMMENTED** [review]
LGTM.

@yangchiu Are we planning to add test cases for node network disconnection when replica rebuilding is in progress?

File: e2e/libs/volume/volume.py
Comment (yangchiu) [reply e2e/libs/volume/volume.py:53]: The original sequence order could fail test cases in some conditions. Like the volume is automatically attached, detached and re-attached in a flash.

**Comment by yangchiu** [comment]
> LGTM.
> 
> @yangchiu Are we planning to add test cases for node network disconnection when replica rebuilding is in progress?

I'll add more test cases after these keyword implementation PRs are merged.

File: e2e/tests/network_disconnect.robot
Comment (yangchiu) [reply e2e/tests/network_disconnect.robot:29]: Yes, it's feasible. But I prefer a test case name explains itself. WDYT @longhorn/qa ?

Comment (chriscchien) [reply e2e/tests/network_disconnect.robot:29]: I'm fine with either.

File: e2e/keywords/network.resource
Comment (c3y1huang) [inline e2e/keywords/network.resource:12]:
```suggestion
Disconnect volume node network of statefulset ${statefulset_id} for ${duration} seconds
```

File: e2e/libs/volume/crd.py
Comment (c3y1huang) [inline e2e/libs/volume/crd.py:217]: Is there a specific reason for the removal of the final assertion? Same below.

**Review by c3y1huang — CHANGES_REQUESTED** [review]

File: e2e/keywords/network.resource
Comment (yangchiu) [reply e2e/keywords/network.resource]: Refined. Thanks!

File: e2e/libs/volume/crd.py
Comment (yangchiu) [reply e2e/libs/volume/crd.py:217]: Refined. Thanks!

**Review by khushboo-rancher — APPROVED** [review]

**Pull request merged**
